kratos/elements/distance_calculation_flux_based_element.cpp

```diff
@@ -0,0 +1,502 @@
+    array_1d<BoundedMatrix<double, 4, 3>, 4>& rDN_DXContainer 
+    )
+{
+    BoundedMatrix<double,4,3> DN_DX; // Gradients matrix
+    array_1d<double,4> N;            // Position of the gauss point
+    double volume;
+    GeometryUtils::CalculateGeometryData(rGeometry, DN_DX, N, volume);
+    GetSimplexShapeFunctionsOnGauss(rNContainer);
+    for (unsigned int i = 0; i < 4; ++i) {
+        rGaussWeights[i] = volume/4.0;
+        rDN_DXContainer[i] = DN_DX;
+    }
+
+    
+}
+
+//generic. lower performance but valid for all geoms.
+template <unsigned int TDim , unsigned int TNumNodes >
+void DistanceCalculationFluxBasedElement<TDim,TNumNodes>::CalculateGaussPointsData(
+    const GeometryType& rGeometry,
+    BoundedVector<double,TNumNodes> &rGaussWeights,
+    BoundedMatrix<double,TNumNodes,TNumNodes> &rNContainer,
+    array_1d<BoundedMatrix<double, TNumNodes, TDim>, TNumNodes>& rDN_DXContainer 
+    )
+{
+    Vector det_j_vector(TNumNodes);
+    // const auto& r_geom = this->GetGeometry();
+    const auto integration_method = GeometryData::IntegrationMethod::GI_EXTENDED_GAUSS_2;
+    rNContainer = rGeometry.ShapeFunctionsValues(integration_method);
+    ShapeFunctionsGradientsType DN_DXContainer_unbounded;
+    //getting information from geometry in DenseVector<Matrix> format
+    rGeometry.ShapeFunctionsIntegrationPointsGradients(DN_DXContainer_unbounded, det_j_vector, integration_method);
+
+
+    const unsigned int number_of_gauss_points = rGeometry.IntegrationPointsNumber(integration_method);
+    const GeometryType::IntegrationPointsArrayType& integration_points = rGeometry.IntegrationPoints(integration_method);
+
+    for (unsigned int g = 0; g < number_of_gauss_points; ++g){
+        rGaussWeights[g] = det_j_vector[g] * integration_points[g].Weight();
+        rDN_DXContainer[g] = DN_DXContainer_unbounded[g];
+    }
+}
+
+
+//////////////////////////////////////////////////////////////////////////////
+// Public Operations
+
+template <unsigned int TDim , unsigned int TNumNodes >
+void DistanceCalculationFluxBasedElement<TDim, TNumNodes >::CalculateLocalSystem(
+    MatrixType &rLeftHandSideMatrix,
+    VectorType &rRightHandSideVector,
+    const ProcessInfo &rCurrentProcessInfo)
+{
+    KRATOS_TRY
+
+    // Resize and intialize output
+    if (rLeftHandSideMatrix.size1() != LocalSize)
+        rLeftHandSideMatrix.resize(LocalSize, LocalSize, false);
+
+    if (rRightHandSideVector.size() != LocalSize)
+        rRightHandSideVector.resize(LocalSize, false);
+
+    noalias(rLeftHandSideMatrix) = ZeroMatrix(LocalSize, LocalSize);
+    noalias(rRightHandSideVector) = ZeroVector(LocalSize);
+
+    const unsigned int step = rCurrentProcessInfo[FRACTIONAL_STEP];
+    
+    if(step == 1){ //solve a transcient diffusion problem
+        CalculatePotentialFlowSystem(rLeftHandSideMatrix, 
+                                     rRightHandSideVector,
+                                     rCurrentProcessInfo);
```

Comment:
```suggestion
        CalculatePotentialFlowSystem(
            rLeftHandSideMatrix, 
            rRightHandSideVector,
            rCurrentProcessInfo);
```
Style guide suggestion. Also happens in other places.

Reply:
ok